packages/vats/tools/protobuf-utils.js

```diff
@@ -0,0 +1,28 @@
+import { decodeBase64, encodeBase64 } from '@endo/base64';
+
+/** @import {Any} from 'cosmjs-types/google/protobuf/any.js' */
+/** @typedef {{ typeUrl: string; valueBase64: string }} Proto3MsgBase64 */
+/**
+ * base64 encode `value` (possibly binary ) for cross-vat communication
+ *
+ * @param {Any} msg
+ * @returns {Proto3MsgBase64}
+ */
```

Comment:
I think this change requires more explicit understanding of why Base64 is required:
- If you declare a Golang struct to have a `[]byte` member, then when you `json.Marshal` or `json.Unmarshal` it, the `encoding/json` Golang module converts `[]byte` to and from base64.
- `AnyKindOfProtoCodec.toJSON(object)`, including `Any.toJSON(object)` returns a "message" with byte arrays converted to Base64
-  `AnyKindOfProtoCodec.fromJSON(message), including `Any.fromJSON(message)` returns an object with Base64 converted to byte arrays
- That, to me, suggests that the type should be named something like `AnyJson` and defined as `{ typeUrl: string, value: string }`, since that is just the Jsonable version of `type Any = { typeUrl: string, value: Uint8Array }`

Reply:
cf cosmic-proto/dist/codegen/google/protobuf/any.js

Reply:
The names `Any` and `AnyJson` make sense to me. I was hoping to distinguish between them structurely by their keys, but I forgot about this precedence that should be maintained: https://github.com/Agoric/agoric-sdk/blob/297395192f437a7626553ca16abd934ea9bc429f/packages/cosmic-proto/src/codegen/google/protobuf/any.ts#L260-L269

Fortunately TS can still distinguish between them. 

I wonder why that method returns `unknown` instead of `{ typeUrl: string, value: string }`. 

The two codecs in this file, we don't need them, right? It's just Any.fromJSON and Any.toJSON. I'll give that a whirl

Reply:
> I wonder why that method returns `unknown`

Maybe because just because the transpiler is being slothful.  A proper `toJSON` return type would need to be prepared to either introduce a new type, or inline the type for an object that could potentially have many properties.

> The two codecs in this file, we don't need them, right? It's just Any.fromJSON and Any.toJSON.

That's correct.

Reply:
> Maybe because just because the transpiler is being slothful.

Seems so. Meanwhile I'll add:
```js
// TODO make codegen toJSON() return these instead of unknown
/**
 * Proto Any with arrays encoded as base64
 */
export type Base64Any<T> = {
  [Prop in keyof T]: T[Prop] extends Uint8Array ? string : T[Prop];
};
```